scripts/d4xx_to_mavlink.py

```diff
@@ -629,28 +627,34 @@ def get_local_ip():
 else:
     send_msg_to_gcs('Nothing to do. Check params to enable something')
     pipe.stop()
-    vehicle.close()
-    print("INFO: Realsense pipe and vehicle object closed.")
+    conn.mav.close()
+    progress("INFO: Realsense pipe and vehicle object closed.")
     sys.exit()
 
-# Listen and extract necessary attitude's data
-vehicle.add_message_listener('ATTITUDE', att_msg_callback)
-# vehicle.add_message_listener('AHRS2', ahrs2_msg_callback)
-
+glib_loop = None
 if RTSP_STREAMING_ENABLE is True:
     send_msg_to_gcs('RTSP at rtsp://' + get_local_ip() + ':' + RTSP_PORT + RTSP_MOUNT_POINT)
     Gst.init(None)
     server = GstServer()
-    threading.Thread(target=GstLoop, args=()).start()
+    glib_loop = GLib.MainLoop()
+    glib_thread = threading.Thread(target=glib_loop.run, args=())
+    glib_thread.start()
 else:
     send_msg_to_gcs('RTSP not streaming')
 
 sched.start()
 
+main_loop_should_quit = False
+def sigint_handler(sig, frame):
```

Comment:
this is nice, it means that someone could "Ctrl-C" the script and make it stop I think.  Are there are interrupt handlers that we need to be careful of?  having the camera stop working is not great in some situations.

Reply:
Only the default interrupt handler is present before adding this patch.

Yes, this is supposed to clean the camera connection a bit and perhaps make it lock up less...